typescript/schedules.md

```diff
@@ -0,0 +1,1255 @@
+    // every hour at minute 5
+    intervals: [
+      {
+        every: '1h',
+        at: '5m',
+      },
+    ],
+    // every 20 days since epoch at day 2
+    // intervals: [{
+    //   every: '20d',
+    //   at: '2d'
+    // }]
+    skip: [
+      {
+        // skip 11:05 pm
+        hour: 23,
+        minute: 5,
+      },
+    ],
+    endAt: addWeeks(new Date(), 4),
+    jitter: '30s',
+    timezone: 'US/Pacific',
+  },
+  action: {
+    workflowId: 'wf-biz-id',
+    type: myWorkflow,
+    args: ['arg1', 'arg2'],
+  },
+  overlap: ScheduleOverlapPolicy.BufferOne,
+  catchupWindow: '2m',
+  pauseOnFailure: true,
+  note: 'started schedule',
+  pause: true, // this sets `state.paused: true` (default false)
+  limitedActions: 10,
+  triggerImmediately: true,
+  backfill: [
+    {
+      start: new Date(),
+      end: new Date(),
+      overlap: ScheduleOverlapPolicy.AllowAll,
+    },
+  ],
+  memo,
+  searchAttributes,
+});
+
+const schedule = client.getHandle('schedule-biz-id');
+
+const scheduleDescription = await schedule.describe();
+
+// For later, pending API finalization:
+// https://github.com/temporalio/proposals/pull/62/files#r933532170
+// const matchingStartTimes = await schedule.listMatchingTimes({ start: new Date(), end: new Date() })
+// const matchingStartTimes = await client.listMatchingTimes({ spec, start, end })
+
+await schedule.update(
+  (schedule) => {
+    schedule.spec.intervals[0].every = '1d';
+    // unset with:
+    // delete schedule.spec.intervals;
+
+    // return false to stop retrying early
+  },
+  { retry: retryPolicy }
+);
+
+await schedule.trigger();
+await schedule.backfill({ startAt: new Date(), endAt: addWeeks(new Date(), 1) }); // also takes array
+await schedule.pause('note: pausing');
+await schedule.unpause('now unpause');
+
+await schedule.delete();
+
+const { schedules, nextPageToken } = await client.listByPage({
+  pageSize: 50,
+  nextPageToken: 'base64',
+});
+
+for await (const schedule: Schedule of client.list()) {
+  const { id, memo, searchAttributes } = schedule;
+  // ...
+}
+```
+
+### Types
+
+```ts
+import { DataConverter, LoadedDataConverter } from '@temporalio/common';
+import { loadDataConverter } from '@temporalio/internal-non-workflow-common';
+import {
+  composeInterceptors,
+  Headers,
+  Replace,
+  RetryPolicy,
+  SearchAttributes,
+  Workflow,
+} from '@temporalio/internal-workflow-common';
+import { temporal } from '@temporalio/proto';
+import os from 'os';
+import { Connection } from './connection';
+import { ScheduleClientCallsInterceptor, ScheduleClientInterceptors } from './interceptors';
+import { ConnectionLike, Metadata, WorkflowService } from './types';
+import { WorkflowHandle, WorkflowStartOptions } from './workflow-client';
+
+import type { NonNegativeInteger, RequireAtLeastOne } from 'type-fest';
+
+// TODO is a non-generic NonNegativeInteger possible? The ones below error due to no type argument
+export type PositiveInteger = Exclude<NonNegativeInteger, 0>;
```

Comment:
I think this is overthinking types. Just use numbers and move on.

Reply:
+1 but we should decide which validations are the client's responsibility and which ones are done only server side.